src/ascent/runtimes/flow_filters/ascent_runtime_blueprint_filters.cpp

```diff
@@ -153,24 +158,51 @@ BlueprintVerify::execute()
         ASCENT_ERROR("blueprint_verify input must be a conduit node");
     }
 
-
     std::string protocol = params()["protocol"].as_string();
 
     Node v_info;
     Node *n_input = input<Node>(0);
-    if(!conduit::blueprint::verify(protocol,
-                                   *n_input,
-                                   v_info))
+    
+    // some MPI tasks may not have data, that is fine
+    // but blueprint verify will fail, so if the
+    // input node is empty skip verify
+    int local_verify_ok = 0;
+    if(!n_input->dtype().is_empty())
     {
-        n_input->print();
-        v_info.print();
-        ASCENT_ERROR("blueprint verify failed for protocol"
-                      << protocol << std::endl
-                      << "details:" << std::endl
-                      << v_info["message"].as_string());
-                      //<< v_info.to_json());
+        if(!conduit::blueprint::verify(protocol,
```

Comment:
I am not sure that this is the correct place to do this. If we have an extract that takes `source` then verify is never called. Two choices: 1) always call verify as a part of `source` or move this logic into `publish`. Are we supporting the case where a rank might not call publish?

Reply:
There are some other checks in publish like consistent domain ids, so if it was not called on every rank then there would be a MPI hang.

Reply:
just to circle back: we aren't supporting cases where publish or execute aren't called on some subset of MPI tasks. All MPI tasks need to call to coordinate. It's ok if some publish empty data. All need to execute the same actions.